internal/pkg/bulk/bulk_test.go

```diff
@@ -272,6 +273,90 @@ func TestCancelCtx(t *testing.T) {
 	}
 }
 
+// verify that child bulker stops when bulker ctx cancelled
+func TestCancelCtxChildBulker(t *testing.T) {
+	bulker := NewBulker(nil, nil)
+
+	ctx, cancelF := context.WithCancel(context.Background())
+	ctx = testlog.SetLogger(t).WithContext(ctx)
+
+	logger := testlog.SetLogger(t)
+	outputMap := make(map[string]map[string]interface{})
+	outputMap["remote"] = map[string]interface{}{
+		"type":          "remote_elasticsearch",
+		"hosts":         []interface{}{"https://remote-es:443"},
+		"service_token": "token1",
+	}
+	childBulker, _, err := bulker.CreateAndGetBulker(ctx, logger, "remote", outputMap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		_, err := childBulker.APIKeyAuth(ctx, apikey.APIKey{})
+
+		if !errors.Is(err, context.Canceled) {
+			t.Error("Expected context cancel err: ", err)
+		}
+	}()
+
+	time.Sleep(time.Millisecond)
+	cancelF()
+
+	wg.Wait()
+}
+
+// verify that old bulker ctx is canceled if cfg changed and a new one is started
+func TestCancelCtxChildBulkerReplaced(t *testing.T) {
+	bulker := NewBulker(nil, nil)
+
+	ctx, _ := context.WithCancel(context.Background())
```

Comment:
we can defer the cancel to make the linter happy, or just disable it for this line